src/controllers/userController.ts

```diff
@@ -52,37 +48,21 @@ export class UserController {
   }
   //Authenticate a user with provided username and password
   public async login(
-    req: Express.Request,
+    req: AuthenticatedBasicRequest,
     res: Express.Response
   ): Promise<Express.Response<{ accessToken: string; refreshToken: string }>> {
-    try {
-      const { username, password } = <loginDto>matchedData(req);
-      const user = <IUser>await userService.getUser(username);
-
-      if (!user) {
-        return res.sendStatus(StatusCodes.NOT_FOUND);
-      }
-
-      const isValid = await userService.comparePassword(
-        password,
-        user.password
-      );
-
-      if (isValid) {
-        const newTokens = await userService.generateToken(user);
-        return res.send(newTokens);
-      } else {
-        return res.status(StatusCodes.UNAUTHORIZED).send('Invalid credentials');
-      }
-    } catch (err) {
-      console.log(err);
-      return res.status(StatusCodes.INTERNAL_SERVER_ERROR);
+    if (!req.user) {

```

Comment:
You should leave try ... catch ... for catching "global" errors within controller scope